src/controller/CHIPDeviceController.cpp

```diff
@@ -250,11 +251,24 @@ void DeviceController::OnLocalNOCChainGeneration(void * context, CHIP_ERROR stat
         err = ConvertX509CertsToChipCertArray(noc, icac, chipCertSpan);
         SuccessOrExit(err);
 
-        err = fabric->SetOperationalCertsFromCertArray(chipCertSpan);
+        err = newFabric.SetOperationalCertsFromCertArray(chipCertSpan);
         SuccessOrExit(err);
     }
 
-    err = controller->mFabrics.Store(fabric->GetFabricIndex());
+    err = controller->mFabrics.AddNewFabric(newFabric, controller->mFabricIndex);
+    SuccessOrExit(err);
+    ChipLogProgress(Controller, "Added new fabric %d", controller->mFabricIndex);
+
+    ChipLogProgress(Controller, "Storing controller credentials");
+    err = controller->mFabrics.Store(controller->mFabricIndex);
+    SuccessOrExit(err);
+
+    fabric = controller->mFabrics.FindFabricWithIndex(controller->mFabricIndex);
+    VerifyOrExit(fabric != nullptr, err = CHIP_ERROR_INCORRECT_STATE);
+
+    err = fabric->GetCredentials(controller->mCredentials, controller->mCertificates, controller->mRootKeyId,
```

Comment:
A fabric doesn't have credentials, but there are credentials for a node associated with a fabric.

A Node joins a Fabric. We do not "add fabrics" to things. We "are part of a fabric because one of our Node operational credentials has refers to a given fabric".

Reply:
Agree, I guess the term might be misused here. The intent is that find the credentials of current node corresponding to the given fabric (if the node has been commissioned on multiple of them). From that point of view, what would be a better name for FabricTable/FabricInfo?

Reply:
`FabricsTable` might= be a good name

Reply:
This code will get reworked as part of fixing #8910.
The rename of `FabricsTable` could be handled in that rework PR.